### PR TITLE
improved Nx conversion performance

### PR DESCRIPTION
**Pull request by BradReesWork**
refactored how to convert from Nx to cuGraph to improve performance

### CODE REVIEW

File: python/cugraph/cugraph/utilities/nx_factory.py [hunk 1]
```diff
@@ -28,7 +29,82 @@
 nx = import_optional("networkx")
 
 
+def convert_unweighted_to_gdf(NX_G):
+    _edges = NX_G.edges(data=False)
+    src = [s for s, _ in _edges]
+    dst = [d for _, d in _edges]
```
Comment (rlratzel) [inline python/cugraph/cugraph/utilities/nx_factory.py:35]: You can do this in one pass using `zip()`
```suggestion
    (src, dst) = zip(*_edges)
```

File: python/cugraph/cugraph/utilities/nx_factory.py [hunk 2]
```diff
@@ -28,7 +29,82 @@
...
+
+    return _gdf
+
+
+def convert_weighted_named_to_gdf(NX_G, weight):
+    _edges = NX_G.edges(data=weight)
+
+    src = [s for s, _, _ in _edges]
+    dst = [d for _, d, _ in _edges]
+    wt = [w for _, _, w in _edges]
```
Comment (rlratzel) [inline python/cugraph/cugraph/utilities/nx_factory.py:49]: same feedback as above re: `zip()`

File: python/cugraph/cugraph/utilities/nx_factory.py [hunk 3]
```diff
@@ -28,7 +29,82 @@
...
+    weight: weight column name. Only used if
+    nxG.is_weighted() is True
+    """
+
+    if isinstance(nxG, nx.classes.digraph.DiGraph):
+        G = cugraph.Graph(directed=True)
+    elif isinstance(nxG, nx.classes.graph.Graph):
+        G = cugraph.Graph()
+    else:
+        raise ValueError(
```
Comment (rlratzel) [inline python/cugraph/cugraph/utilities/nx_factory.py:85]: This should probably be a `TypeError`

File: python/cugraph/cugraph/utilities/nx_factory.py (hunk 1 above)
Comment (BradReesWork) [reply python/cugraph/cugraph/utilities/nx_factory.py:35]: I tried that and you can do it in one pass but the performance is not as good

Comment (BradReesWork) [reply python/cugraph/cugraph/utilities/nx_factory.py:35]: two pass takes 500ms,  the zip approach take 1.2 sec

Comment (rlratzel) [reply python/cugraph/cugraph/utilities/nx_factory.py:35]: ah ok, I'll just mark as resolved then.

File: python/cugraph/cugraph/utilities/nx_factory.py (hunk 3 above)
Comment (BradReesWork) [reply python/cugraph/cugraph/utilities/nx_factory.py:85]: good catch

**Review by rlratzel — APPROVED** [review]

**Comment by BradReesWork** [comment]
rerun tests

**Comment by rlratzel** [comment]
rerun tests

**Comment by BradReesWork** [comment]
@gpucibot merge

**Pull request merged**